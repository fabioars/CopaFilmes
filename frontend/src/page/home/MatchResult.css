/* MatchResult */
.MatchResult__item {
    margin-bottom: 16px;
    min-height: 144px;
    display: flex;
    align-items: center;
    background-color: #fff;
}
.MatchResult__position {
    display: flex;
    align-items: center;
    justify-content: center;
    min-width: 120px;
    align-self: stretch;
    font-size: 48px;
    text-align: center;
    color: #fff;
    background-color: var(--bg-color-darken);
}
.MatchResult__movie {
    font-size: 22px;
    padding: 20px 30px;
}
