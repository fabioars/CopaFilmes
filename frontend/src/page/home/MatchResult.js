import React from 'react';
import './MatchResult.css';

export default function MatchResult({ match }) {
    return (
        <section className="MatchResult">
            <div className="MatchResult__item">
                <span className="MatchResult__position">1ยบ</span>
                <span className="MatchResult__movie">{match.first.titulo}</span>
            </div>

            <div className="MatchResult__item">
                <span className="MatchResult__position">2ยบ</span>
                <span className="MatchResult__movie">{match.second.titulo}</span>
            </div>
        </section>
    );
}
