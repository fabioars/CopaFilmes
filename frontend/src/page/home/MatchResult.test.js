import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';
import MatchResult from './MatchResult';

describe('Movie Counter', () => {
    let container = null;
    let match = null;

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);

        match = {
            first:{
                id: "tt4154756",
                titulo: "Vingadores: Guerra Infinita",
                ano: 2018,
                nota: 8.8
            },
            second:{
                id: "tt3606756",
                titulo: "Os IncrÃ­veis 2",
                ano: 2018,
                nota: 8.5
            }
        };
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        container.remove();
        container = null;
        match = null;
    });

    it('renders correctly', () => {
        const tree = renderer.create(<MatchResult match={match} />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
