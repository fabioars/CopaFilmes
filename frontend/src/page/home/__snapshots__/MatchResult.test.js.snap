// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Counter renders correctly 1`] = `
<section
  className="MatchResult"
>
  <div
    className="MatchResult__item"
  >
    <span
      className="MatchResult__position"
    >
      1º
    </span>
    <span
      className="MatchResult__movie"
    >
      Vingadores: Guerra Infinita
    </span>
  </div>
  <div
    className="MatchResult__item"
  >
    <span
      className="MatchResult__position"
    >
      2º
    </span>
    <span
      className="MatchResult__movie"
    >
      Os Incríveis 2
    </span>
  </div>
</section>
`;
